#
# Temporary fix, to use OpenLDAP library, instead of legacy Mozilla LDAP
# Hardcodes -lldap_r-2.4 in configure script, instead of -lldap.
# Necessary to build current krb5, otherwise:
#         kdb_ldap_conn.c, line 142: undefined symbol: LDAP_SASL_QUIET
#
# Reported upstream:
#    http://krbdev.mit.edu/rt/Ticket/Display.html?id=8261
# Patch source: in-house
#
diff -ur old/src/configure.in new/src/configure.in
--- old/src/configure.in	2013-11-04 16:55:08.000000000 -0800
+++ new/src/configure.in	2014-05-22 04:28:42.013736461 -0700
@@ -1118,20 +1118,20 @@
 ldap_lib=""
 if test -n "$OPENLDAP_PLUGIN"; then
   AC_CHECK_HEADERS(ldap.h lber.h, :, [AC_MSG_ERROR($ac_header not found)])
-  AC_CHECK_LIB(ldap, ldap_init, :, [AC_MSG_ERROR(libldap not found or missing ldap_init)])
+  AC_CHECK_LIB(ldap_r-2.4, ldap_init, :, [AC_MSG_ERROR(libldap not found or missing ldap_init)])
   old_LIBS="$LIBS"
-  LIBS="$LIBS -lldap"
+  LIBS="$LIBS -lldap_r-2.4"
   AC_CHECK_FUNCS(ldap_initialize ldap_url_parse_nodn ldap_unbind_ext_s ldap_str2dn ldap_explode_dn)
   LIBS="$old_LIBS"
 
   BER_OKAY=0
-  AC_CHECK_LIB(ldap, ber_init, [BER_OKAY=1])
+  AC_CHECK_LIB(ldap_r-2.4, ber_init, [BER_OKAY=1])
   if test "$BER_OKAY" = "1"; then 
-    LDAP_LIBS='-lldap'
+    LDAP_LIBS='-lldap_r-2.4'
   else
-    AC_CHECK_LIB(lber, ber_init, [BER_OKAY=1], [AC_MSG_WARN([libber not found])])
+    AC_CHECK_LIB(lber-2.4, ber_init, [BER_OKAY=1], [AC_MSG_WARN([libber not found])])
     if test "$BER_OKAY" = "1"; then 
-      LDAP_LIBS='-lldap -llber'
+      LDAP_LIBS='-lldap_r-2.4 -llber-2.4'
     else
       AC_ERROR("BER library missing - cannot build LDAP database module")
     fi
